File: 프로그래머스/0/120839. 가위 바위 보/가위 바위 보.py
def solution(rsp):
    # 변환 규칙을 딕셔너리로 정의
    win_dict = {'0': '5', '2': '0', '5': '2'}
    
    # 리스트 컴프리헨션을 사용하여 변환된 값을 저장 (성능 최적화)
    answer = [win_dict[r] for r in rsp]
    
    # 리스트를 문자열로 변환하여 반환
    return ''.join(answer)

# 테스트 실행
print(solution("2"))     # 출력: "0"
print(solution("205"))   # 출력: "052"

File: 프로그래머스/0/120839. 가위 바위 보/가위 바위 보.java
import java.util.*;

public class Solution {

    public static String solution(String rsp) {
        // 1. 이기는 규칙을 Map으로 정의
        Map<Character, Character> winMap = new HashMap<>();
        winMap.put('0', '5');  // 바위(0)는 보(5)를 이김
        winMap.put('2', '0');  // 가위(2)는 바위(0)를 이김
        winMap.put('5', '2');  // 보는 가위(2)에게 짐 → 가위가 이김

        // 2. 결과를 저장할 StringBuilder
        StringBuilder result = new StringBuilder();

        // 3. 각 문자에 대해 변환
        for (int i = 0; i < rsp.length(); i++) {
            char c = rsp.charAt(i);
            result.append(winMap.get(c));
        }

        // 4. 결과 문자열 반환
        return result.toString();
    }

    // ✅ 테스트 실행
    public static void main(String[] args) {
        System.out.println(solution("2"));     // 출력: 0
        System.out.println(solution("205"));   // 출력: 052
    }
}
